egs/wsj/s5/steps/online/nnet2/train_ivector_extractor.sh

```diff
@@ -184,3 +184,9 @@ done
 
 rm $dir/final.ie 2>/dev/null
 ln -s $x.ie $dir/final.ie
+
+# assign a unique id to this extractor
+extractor_id_file=$(find $srcdir/ -name "extractor_id_*")
+[ `echo $extractor_id_file | wc -w` -gt 0 ] && rm $extractor_id_file;
+extractor_id_file=$dir/extractor_id_$(data + '%m_%d_%H_%M')
```

Comment:
The way you are doing it makes the scripting unnecessarily hard, and casual inspection by script more difficult.
I think you should do it like:

date +'%m_%d_%H_%M' >$dir/extractor_id

and then use cmp to compare them when needed.
Also your way of doing it is against any pattern we currently use.